tests/test_zephyr_interface_no_ci.py

```diff
@@ -0,0 +1,220 @@
+            "values",
+        ]
+
+        statuses = await self.zapi.get_statuses()
+        self.assertListEqual(list(statuses.keys()), payload_expected_keys)
+
+    @pytest.mark.asyncio
+    async def test_get_test_case(self):
+
+        payload_expected_keys = [
+            "id",
+            "key",
+            "name",
+            "project",
+            "createdOn",
+            "objective",
+            "precondition",
+            "estimatedTime",
+            "labels",
+            "component",
+            "priority",
+            "status",
+            "folder",
+            "owner",
+            "testScript",
+            "customFields",
+            "links",
+        ]
+
+        test_case_key = "BLOCK-T21"
+        test_case = await self.zapi.get_test_case(test_case_key)
+        self.assertEqual(test_case["key"], test_case_key)
+        self.assertListEqual(list(test_case.keys()), payload_expected_keys)
+
+    async def test_get_test_case_steps(self):
+
+        payload_expected_keys = [
+            "next",
+            "startAt",
+            "maxResults",
+            "total",
+            "isLast",
+            "values",
+        ]
+        payload_value_keys = ["inline", "testCase"]
+
+        test_case_key = "BLOCK-T21"
+        test_case_steps = await self.zapi.get_test_case_steps(test_case_key)
+        self.assertListEqual(list(test_case_steps.keys()), payload_expected_keys)
+        self.assertListEqual(
+            list(test_case_steps["values"][0].keys()), payload_value_keys
+        )
+
+    async def test_get_test_execution(self):
+
+        payload_expected_keys = [
+            "id",
+            "key",
+            "project",
+            "testCase",
+            "environment",
+            "jiraProjectVersion",
+            "testExecutionStatus",
+            "actualEndDate",
+            "estimatedTime",
+            "executionTime",
+            "executedById",
+            "assignedToId",
+            "comment",
+            "automated",
+            "testCycle",
+            "customFields",
+            "links",
+        ]
+
+        test_execution_id = "BLOCK-E192"
+        test_execution = await self.zapi.get_test_execution(test_execution_id)
+        self.assertListEqual(list(test_execution.keys()), payload_expected_keys)
+        self.assertEqual(test_execution["key"], test_execution_id)
+
+    async def test_extract_test_case_from_test_execution(self):
+
+        expected_test_case_name = "BLOCK-T21"
+        expected_test_case_version = "1"
+
+        test_execution_id = "BLOCK-E192"
+        test_execution = await self.zapi.get_test_execution(test_execution_id)
+        test_case_name, test_case_version = (
+            self.zapi.extract_test_case_from_test_execution(test_execution)
+        )
+
+        self.assertEqual(test_case_name, expected_test_case_name)
+        self.assertEqual(test_case_version, expected_test_case_version)
+
+    async def test_get_test_executions(self):
```

Comment:
Just for the sake of better code ordering, I recommend to move this test above `test_extract_test_case_from_test_execution`.

Reply:
Hmmm... in which sense? I tried to have the methods inside the test class organized alphabetically.

Reply:
Sorry, dismiss this comment. At first glance I thought this test should have been placed above because its structure is similar to the first tests above (like defining the payload and then calling some functions), however ordering alphabetically is good! Sorry for the noise :pray:.